src/pages/home/report/ReportActionsView.js

```diff
@@ -205,6 +211,12 @@ class ReportActionsView extends React.Component {
     }
 
     componentDidUpdate(prevProps) {
+        // Update the last read action for the report currently in view when report data finishes loading.
+        // This report should now be up-to-date and since it is in view we mark it as read.
+        if (!this.props.isLoadingReportData && prevProps.isLoadingReportData) {
+            this.updateNewMarkerAndMarkReadOnce();
+        }
+
```

Comment:
I think the delay is caused as `isLoadingReportData` waits for all reports data to load. On the other hand, new messages are already received on the next if block. 
 
 we should set the new marker when the new messages are received as well which happens in https://github.com/Expensify/App/blob/1d72a5e01ed6e0c0b500f7e0617d05eb8c05dee9/src/pages/home/report/ReportActionsView.js#L216. 
 
 but if you call `Report.updateLastReadActionID(this.props.reportID);` inside this nothing happens. 
 this function waits on the `isLoadingReportData` before marking the last read https://github.com/Expensify/App/blob/1d72a5e01ed6e0c0b500f7e0617d05eb8c05dee9/src/libs/actions/Report.js#L1312. 
 
 Thus I think if we can set the `lastReadSequenceNumbers[reportID]` when each new action is added then it will work on time.

Reply:
I got your point @parasharrajat . I agree with you when you said:
> Thus I think if we can set the lastReadSequenceNumbers[reportID] when each new action is added then it will work on time.

But I don't know how we could set `lastReadSequenceNumbers[reportID]`.
As you said, we call `Report.updateLastReadActionID(this.props.reportID)` and nothings happens because we are waiting for report data `(isReportDataLoading === true)`.

I think there's no way to set `lastReadSequenceNumbers[reportID]` within `Report.updateLastReadActionID(this.props.reportID)` because we don't have data to do that, we are just passing `this.props.reportID` as param, and if we add `this.props.report` as params, in that moment it will give us the old report data because report data was not fetched yet and we need report data to know `unreadActionCount`.

Reply:
I'm doing some changes on code and testing it.
I'm going to tell what I think we should do soon.